src/build_test.cc

```diff
@@ -2175,7 +2175,7 @@ TEST_F(BuildWithDepsLogTest, DepFileOKDepsLog) {
 
     Edge* edge = state.edges_.back();
 
-    state.GetNode("bar.h", 0)->MarkDirty();  // Mark bar.h as missing.
+    state.GetNode(std::string("bar.h"), 0)->MarkDirty();  // Mark bar.h as missing.
```

Comment:
Why is this an improvement?

The c-style string will be converted to an std::string if needed, and not, if not.

Why insert an explicit conversion?

Reply:
There is no intention to improve the tests. In order not to change the tests we have to remove the second GetNode candidate or add one that explicitly accepts a char *

Reply:
Why doesn't it already accept a char*? it does currently.

Because of your change to only accept string&&, in a variety of functions?

I strongly disagree with changing the parameter to those functions to string&&.